src/components/Rewards/ChooseReward/ChooseReward.js

```diff
@@ -0,0 +1,110 @@
+import {getUserDatabase} from "../../../functions/UserData";
+import {useHistory} from "react-router-dom";
+import './ChooseReward.scss';
+import {useEffect, useState} from "react";
+import placeholderImage from "../../../assets/images/placeholder-image.jpg"
+import {database} from '../../../firebase';
+import {ref, child, get} from 'firebase/database';
+
+const ChooseReward = () => {
+    // Get the current player
+    const currentUser = getUserDatabase();
+    const history = useHistory();
+    /*
+     Sample data, once again we need to make this empty once we get the partners in the database
+     */
+    const [partners, setPartners] = useState([]);
+
+    useEffect(() => {
+        // Get the partner data from the database
+        const getDataFromDB = async () => {
+            const dbRef = ref(database);
+            get(child(dbRef, `partners`)).then((snapshot) => {
+                const data = snapshot.val();
+                setPartners(Object.values(data));
+            });
+        }
+
+        getDataFromDB().then();
+    }, [])
+
+    const [selected, setSelected] = useState(null);
+
+    // Calculate if the user is eligible for the challenge
+    const isEligible = (user) => {
+        if (!user.badges) return false;
+        return user.completedActionForChallenger || user.badges.length >= 8;
+    }
+
+    if (!isEligible(currentUser)) {
+        history.push('/progress');
```

Comment:
instead, push them to the /signup page (that will push them to /progress or /actions depending on user type)